File: dev-docs/snippets/documents/list-group-permissions.swift
import JsBaoClient

// List group-based permissions on a document as typed
// `[DocumentGroupPermissionEntry]`. Pass `includeSystem: true` to surface
// platform-managed internal groups (admin tooling).
func listGroupPermissions(client: JsBaoClient, documentId: String) async throws {
  // #region example
  let groups = try await client.documents.listGroupPermissions(
    documentId: documentId, includeSystem: false
  )
  let firstGroupId = groups.first?.groupId
  // #endregion example
  _ = firstGroupId
}

File: dev-docs/snippets/documents/list-group-permissions.ts
import type { JsBaoClient } from "js-bao-wss-client";

// List group-based permissions on a document. Pass `{ includeSystem: true }`
// to surface platform-managed internal groups (admin tooling).
export async function listGroupPermissions(
  client: JsBaoClient,
  documentId: string,
) {
  // #region example
  const groups = await client.documents.listGroupPermissions(documentId, {
    includeSystem: false,
  });
  // #endregion example
  return groups;
}
